src/__tests__/test-cache-service.ts

```diff
@@ -0,0 +1,141 @@
+				);
+			}
+		},
+		del(
+			call: ServerUnaryCall<DelRequest, DelResponse>,
+			callback: sendUnaryData<DelResponse>
+		): void {
+			const cacheName = call.request.getProject() + "_" + call.request.getName();
+			if (TestCacheService.CACHE_MAP.has(cacheName)) {
+				if (TestCacheService.CACHE_MAP.get(cacheName).has(call.request.getKey())) {
+					TestCacheService.CACHE_MAP.get(cacheName).delete(call.request.getKey());
+					callback(
+						undefined,
+						new DelResponse().setStatus("deleted").setMessage("Deleted key count# 1")
+					);
+				}
+			} else {
+				callback(new Error("cache does not exist"), undefined);
+			}
+		},
+		deleteCache(
+			call: ServerUnaryCall<DeleteCacheRequest, DeleteCacheResponse>,
+			callback: sendUnaryData<DeleteCacheResponse>
+		): void {
+			const cacheName = call.request.getProject() + "_" + call.request.getName();
+			if (TestCacheService.CACHE_MAP.has(cacheName)) {
+				TestCacheService.CACHE_MAP.delete(cacheName);
+				callback(
+					undefined,
+					new DeleteCacheResponse().setStatus("deleted").setMessage("Deleted cache")
+				);
+			} else {
+				callback(new Error("cache does not exist"), undefined);
+			}
+		},
+		get(
+			call: ServerUnaryCall<GetRequest, GetResponse>,
+			callback: sendUnaryData<GetResponse>
+		): void {
+			const cacheName = call.request.getProject() + "_" + call.request.getName();
+			if (TestCacheService.CACHE_MAP.has(cacheName)) {
+				if (TestCacheService.CACHE_MAP.get(cacheName).has(call.request.getKey())) {
+					const value = TestCacheService.CACHE_MAP.get(cacheName).get(call.request.getKey());
+					callback(undefined, new GetResponse().setValue(value));
+				} else {
+					callback(new Error("cache key does not exist"), undefined);
+				}
+			} else {
+				callback(new Error("cache does not exist"), undefined);
+			}
+		},
+		keys(
+			call: ServerUnaryCall<KeysRequest, KeysResponse>,
+			callback: sendUnaryData<KeysResponse>
+		): void {
+			const cacheName = call.request.getProject() + "_" + call.request.getName();
+			if (TestCacheService.CACHE_MAP.has(cacheName)) {
+				const result: Array<string> = new Array<string>();
+				for (let key of TestCacheService.CACHE_MAP.get(cacheName).keys()) {
+					result.push(key);
+				}
+				callback(undefined, new KeysResponse().setKeysList(result));
+			} else {
+				callback(new Error("cache does not exist"), undefined);
+			}
+		},
+		listCaches(
+			call: ServerUnaryCall<ListCachesRequest, ListCachesResponse>,
+			callback: sendUnaryData<ListCachesResponse>
+		): void {
+			const result: Array<CacheMetadata> = new Array<CacheMetadata>();
+			for (let key of TestCacheService.CACHE_MAP.keys()) {
+				if (key.startsWith(call.request.getProject()))
+					result.push(
+						new CacheMetadata().setName(key.replace(call.request.getProject() + "_", ""))
+					);
+			}
+			callback(undefined, new ListCachesResponse().setCachesList(result));
+		},
+		set(
```

Comment:
Do you have a test for testing with options as well?

Reply:
we don't yet support all options. I am going to iterate on options to bring them in sync with api, server and sdks.